### PR TITLE
[BUGFIX] Build subarray if found in regex, regardless of other matches

### PR DESCRIPTION
**Pull request by smichaelsen**
To solve #449, we can change the order of result handling. Tests were added, which test for the existing behavior.

### CODE REVIEW

**Comment by smichaelsen** [comment]
> @dlubitz: This would solve the parser output in this case, but not the issue itself. But if it's the only way ATM to solve this, I would also go with that "hacky" solution.

The underlying problem can most likely not be fixed within fluid itself, because it seems to be somewhere in the PHP environment. As matter of fact there are those environments out there where the marching works differently, so I think Fluid should be able to cope with that case.

File: src/Core/Parser/TemplateParser.php [hunk 1]
```diff
@@ -698,16 +698,16 @@ protected function recursiveArrayHandler($arrayText)
         if (preg_match_all(Patterns::$SPLIT_PATTERN_SHORTHANDSYNTAX_ARRAY_PARTS, $arrayText, $matches, PREG_SET_ORDER)) {
             foreach ($matches as $singleMatch) {
                 $arrayKey = $this->unquoteString($singleMatch['Key']);
-                if (!empty($singleMatch['VariableIdentifier'])) {
+                if (array_key_exists('Subarray', $singleMatch) && !empty($singleMatch['Subarray'])){
```
Comment (NamelessCoder) [inline src/Core/Parser/TemplateParser.php]: I'm not sure this changes anything - and if it does, it merely hides the fact that the array was not detected correctly by PCRE. `$singleMatch['Subarray']` will still be set but will be null or an empty array, which is the core symptom of the problem. So this condition will still not trigger - and if it triggers with anything other than an array, it leads to new PHP errors (constructor of ArrayNode does not accept non-arrays).

Comment (smichaelsen) [reply src/Core/Parser/TemplateParser.php]: > it merely hides the fact that the array was not detected correctly by PCRE.
Yes, that is the point, see my comment above.

In our tests `VariableIdentifier` or `QuotedString` were set *additionally* to `Subarray` and that's why the wrong if/elseif-branch was executed. Changing the order did fix that. I agree that the check for the Subarray is not solid enough, an `is_array()` should be added.

**Comment by albe** [comment]
@NamelessCoder friendly ping regarding reopening and merging this as of your comment on #506 :)

**Comment by NamelessCoder** [comment]
Merged since I can't provoke any errors associated with the change - but I still have this nagging sensation that this could break something. I'd appreciate if you could do an extremely careful round of testing on `dev-master` on both affected and unaffected systems, before I go ahead and backport + create releases.

**Comment by NamelessCoder** [comment]
Version 2.6.9 has been released, if you are affected by this issue and prefer to use tagged release versions, you can now update `typo3fluid/fluid` to fix the issue.

**Pull request merged**